pkg/certificates/endpoint.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package certificates
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/peer"
+
+	"storj.io/storj/pkg/certificates/authorizations"
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+)
+
+// Endpoint implements pb.CertificatesServer
+type Endpoint struct {
+	log             *zap.Logger
+	ca              *identity.FullCertificateAuthority
+	authorizationDB *authorizations.DB
+	minDifficulty   uint16
+}
+
+// NewCertificatesServer creates a new certificate signing grpc server
+func NewCertificatesServer(log *zap.Logger, ident *identity.FullIdentity, ca *identity.FullCertificateAuthority, authorizationDB *authorizations.DB, minDifficulty uint16) *Endpoint {
+	return &Endpoint{
+		log:             log,
+		ca:              ca,
+		authorizationDB: authorizationDB,
+		minDifficulty:   minDifficulty,
+	}
+}
+
+// Sign signs the CA certificate of the remote peer's identity with the `certs.ca` certificate.
+// Returns a certificate chain consisting of the remote peer's CA followed by the CA chain.
+func (endpoint Endpoint) Sign(ctx context.Context, req *pb.SigningRequest) (_ *pb.SigningResponse, err error) {
```

Comment:
The error returned by this function must be gRPC error  https://godoc.org/google.golang.org/grpc#Errorf

Codes: https://github.com/grpc/grpc/blob/master/doc/statuscodes.md

Reply:
I'm realizing that we have to update the documentation https://github.com/storj/docs/blob/master/code/Endpoint.md#grpc-implementation